CONTRIBUTING.md

```diff
@@ -36,6 +36,8 @@ If working on a new component, consider starting it in the `experimental/` direc
 
 If working on an improvement to an existing component that you wish to split into multiple smaller pieces, consider hiding it under the `"experimental"` feature in the crate. Doing so gives a signal to users and tooling that the code is not yet production-ready. Once finished, the `"experimental"` feature can be removed from the crate.
 
+When adding non-experimental code to main, also update the "unreleased" section of the changelog. This simplifies releases and gives us a better overview of what unreleased features we have.
```

Comment:
a thing that we do in Clippy is that we require the PR to have a `changelog: ` in its body, which you can set to `changelog: none` but in general you can fill in whatever you need, and we aggregate them later. That's an alternate way to do this, with less burden on PRs but then more burden on the release process

this seems ok though, provided reviewers enforce this well

Reply:
I think most PRs should be gated changes, very few PRs go directly to stable, so it's those and the stabilization PRs that need to touch the changelog

Reply:
Definitely a good step in the right direction!